compiler/rustc_mir/src/transform/coverage/mod.rs

```diff
@@ -37,6 +37,7 @@ use rustc_span::{CharPos, ExpnKind, Pos, SourceFile, Span, Symbol};
 /// A simple error message wrapper for `coverage::Error`s.
 #[derive(Debug)]
 struct Error {
+    #[allow(dead_code)] // inserted by #85200; maybe remove this whole struct?
     message: String,
```

Comment:
you can this whole struct here :laughing: cc @richkadel

Reply:
I don't understand this comment, sorry!

I didn't remove this struct because it is used in various places as the `Err` component of `Result`, and I wasn't sure what to replace it with (just `()`? An empty `Error` struct? `Option` instead of `Result`?).

Reply:
probably change the functions to be non fallible.

If the function never errors it probably doesn't make much sense to keep returning `Result`

Reply:
But the functions _can_ fail, it seems; the error value just never escapes this module:

https://github.com/rust-lang/rust/blob/491cf5561efb1f5ff33c3234ccd0bc5cacbebe3e/compiler/rustc_mir/src/transform/coverage/mod.rs#L265-L267

Perhaps we should keep the `message` field then if it is useful for the `bug!()` output (or just change it to print `e.message` instead of `e`, which would also make the "field never read" warning go away).

Reply:
I don't understand why this attribute is needed. The `coverage::Error` class is used inside the `coverage` module, and `message` is read (AFAICT), and if this somehow needs to be annotated as `dead_code` when it isn't dead code or unread, seems like a hack. It's so trivial that I would assume whatever change might require this would break a lot of other Rust code.

Regarding the comment, I disagree. I don't think it should be changed or dropped. I should be allowed to have a coverage-specific error type (with a message or any other field). This is specifically typed to ensure the `?` try operator applies only to specific coverage function results.

Let me know if I'm missing something.

Reply:
This is an interesting one. The `Debug` is being used here to convey information...hmm.